float32 distance
uint8 intensity
